### PR TITLE
Add waitAny wrapper and bindings

### PR DESCRIPTION
**Pull request by stopmosk**
Add Python bindings for VideoCapture::waitAny #21826

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/videoio/include/opencv2/videoio.hpp [hunk 1]
```diff
@@ -967,6 +967,12 @@ class CV_EXPORTS_W VideoCapture
             CV_OUT std::vector<int>& readyIndex,
             int64 timeoutNs = 0);
 
+    CV_WRAP static
+    std::vector<int> waitAny(
```
Comment (VadimLevin) [inline modules/videoio/include/opencv2/videoio.hpp:971]: This overload is unnecessary.

```cxx
CV_WRAP static waitAny(const std::vector<VideoCapture>& streams,
                       CV_OUT std::vector<int>& readyIndex,
                       int64_t timeoutNs = 0);
``` 
should work if you add missing conversion function for `int64_t`
```cxx
template<> bool pyopencv_to(PyObject* obj, int64_t& value, const ArgInfo& info);
```

and you should add an appropriate test for it to `modules/python/test/test_misc.py`

File: modules/videoio/misc/python/pyopencv_videoio.hpp [hunk 2]
```diff
@@ -20,4 +21,15 @@ bool pyopencv_to(PyObject *o, std::vector<cv::VideoCaptureAPIs>& apis, const Arg
   return pyopencvVecConverter<cv::VideoCaptureAPIs>::to(o, apis, info);
 }
 
+template<> bool pyopencv_to(PyObject* obj, cv::VideoCapture& stream, const ArgInfo& info)
+{
+    Ptr<VideoCapture> * obj_getp = 0;
```
Comment (VadimLevin) [inline modules/videoio/misc/python/pyopencv_videoio.hpp:26]: Use `nullptr` instead of `0`

File: modules/videoio/misc/python/pyopencv_videoio.hpp [hunk 3]
```diff
@@ -20,4 +21,15 @@ bool pyopencv_to(PyObject *o, std::vector<cv::VideoCaptureAPIs>& apis, const Arg
...
+    if (!pyopencv_VideoCapture_getp(obj, obj_getp))
+        return failmsgp("Incorrect type of self (must be 'VideoCapture' or its derivative)");
+    //Ptr<VideoCapture> self = *(obj_getp);
```
Comment (VadimLevin) [inline modules/videoio/misc/python/pyopencv_videoio.hpp:29]: Remove debug comments

**Review by VadimLevin — CHANGES_REQUESTED** [review]

File: modules/videoio/include/opencv2/videoio.hpp (hunk 1 above)
Comment (stopmosk) [reply modules/videoio/include/opencv2/videoio.hpp:971]: Overload with `vector<int>` return us useful (see #21826 `waitAny()` signature proposal). But I can use old signature, if needed.

Anyway, I'll add missing conversion for `int64_t` and test for it.

Comment (VadimLevin) [reply modules/videoio/include/opencv2/videoio.hpp:971]: The signature will be:
```python
ret_code, indices = cv2.VideoCapture.waitAny(captures)
if ret_code:
    ...
```

Not the best looking API, but it is consistent with all functions having `CV_OUT` arguments.

Comment (stopmosk) [reply modules/videoio/include/opencv2/videoio.hpp]: I have an isuue with int64 test. 

I added `dumpInt64` function to `modules/core/include/opencv2/core/bindings_utils.hpp`

```cpp
CV_WRAP static inline
String dumpInt64(int64 argument)
{
    return cv::format("Int: %ld", argument);
}
```

But on some systems int64 is long and on other systems int64 is long long. 

How can I prorperly handle it?

Comment (VadimLevin) [reply modules/videoio/include/opencv2/videoio.hpp]: Same thing with `size_t`. It was handled with `std::ostringstream`

File: .gitignore [hunk 4]
```diff
@@ -25,3 +25,5 @@ build
 node_modules
 CMakeSettings.json
 xcuserdata/
+/cmake-build-debug/
```
Comment (VadimLevin) [inline .gitignore:28]: This change is useful, but out of scope for this PR

**Review by VadimLevin — COMMENTED** [review]
LGTM, but let's keep .gitignore unchanged

File: .gitignore (hunk 4 above)
Comment (stopmosk) [reply .gitignore]: Oh, I add it by accident. Sure.

**Review by VadimLevin — APPROVED** [review]

**Pull request merged**